client/src/components/Schedule/TableView.tsx

```diff
@@ -11,92 +13,254 @@ import {
-      render: dateWithTimeZoneRenderer(timeZone, 'MMM Do YYYY'),
-      sorter: dateSorter('dateTime'),
-      sortDirections: ['descend', 'ascend'],
-      defaultSortOrder: 'descend',
-    },
-    { title: 'Time', width: 60, dataIndex: 'dateTime', render: dateWithTimeZoneRenderer(timeZone, 'HH:mm') },
-    {
-      title: 'Type',
-      width: 120,
-      dataIndex: ['event', 'type'],
-      render: (value: keyof typeof EventTypeColor) => renderTag(EventTypeToName[value] || value, EventTypeColor[value]),
-    },
-    {
-      title: 'Special',
-      width: 150,
-      dataIndex: 'special',
-      render: (tags: string) => !!tags && tagsRenderer(tags.split(',')),
-    },
-    {
-      title: 'Name',
-      dataIndex: ['event', 'name'],
-      render: (value: string) => <Text strong>{value}</Text>,
-      ...getColumnSearchProps('event.name'),
-    },
-    {
-      title: 'Url',
-      width: 30,
-      dataIndex: ['event', 'descriptionUrl'],
-      render: urlRenderer,
-    },
-    { title: 'Duration', width: 60, dataIndex: 'duration' },
-    {
-      title: 'Organizer',
-      width: 140,
-      dataIndex: ['organizer', 'githubId'],
-      render: (value: string) => !!value && <GithubUserLink value={value} />,
-      ...getColumnSearchProps('organizer.githubId'),
-    },
-    {
-      title: 'Place',
-      dataIndex: 'place',
-      render: placeRenderer,
-      onCell: () => {
-        return {
-          style: {
-            whiteSpace: 'nowrap',
-            maxWidth: 250,
-          },
-        };
+    editable: true,
+  },
+];
+
+export function TableView({ data, timeZone, isAdmin, courseId }: Props) {
+  const [form] = Form.useForm();
+  const [localData, setLocalData] = useState(data);
+  const [editingKey, setEditingKey] = useState('');
+  const courseService = useMemo(() => new CourseService(courseId), [courseId]);
+
+  const isEditing = (record: CourseEvent) => record.id.toString() === editingKey;
+
+  const edit = (record: CourseEvent) => {
+    form.setFieldsValue({
+      ...record,
+      dateTime: moment(record.dateTime),
+      time: moment(record.dateTime),
+      special: record.special ? record.special.split(',') : [],
+    });
+    setEditingKey(record.id.toString());
+  };
+
+  const handleDelete = async (id: number) => {
+    const newData = localData.filter(event => event.id !== id);
+
+    setLocalData(newData);
+    try {
+      await courseService.deleteCourseEvent(id);
+    } catch {
+      message.error('Failed to delete item. Please try later.');
+    }
+  };
+
+  const cancel = () => {
+    setEditingKey('');
+  };
+
+  const save = async (key: React.Key) => {
+    const updatedRow = (await form.validateFields()) as ScheduleRow;
+    const index = localData.findIndex(item => key === item.id.toString());
+
+    if (index > -1) {
+      const newData = [...localData];
+      const editableEvent = newData[index];
+
+      updateEvent(editableEvent, updatedRow);
+      setLocalData(newData);
```

Comment:
Having a few sources of truth brings about issues and complicates the development. Let's go with the approach already used in rs app: /Users/Yuliya_Haluza/work/rsschool-api/client/src/pages/course/admin/events.tsx

cc: @not-SAINT

Reply:
An example of editing a table: /Users/Yuliya_Haluza/work/rsschool-api/client/src/pages/admin/events.tsx

Probably this approach will be even easier than inline editing.